### ISSUE TITLE
How to use different Versions of poetry?

### ISSUE THREAD

**Issue by MaKaNu** [python-poetry/poetry#8811]
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.

## Issue
I am usually on the edge and utilize poetry 1.7.1 at the moment.

Recently I needed to lower my support for a package down to 3.6.9. I actually don't want to, but some old ubuntu 2018.04 forced me.

By doing so I realized 1.7.1 does not support 3.6.9. So I installed 1.2.0a2, the most recent Version available for 3.6.9 it seems.

So for the moment the newest release is not available for me. I use pyenv to control the python versions I have in use.

What is the best practice to have access to both versions?

**Comment by dimbleby** [comment]
Best practice is to read the docs, which discuss this.

Please close.

**Comment by MaKaNu** [comment]
@dimbleby 
Are you talking about Managing environments?

If yes this was not a solution for my problem.

If I have installed python3.6 via pyenv and selected 3.6 I get an Error for the venv creation:

```bash
SyntaxError: future feature annotations is not defined
```
which is reasoanble since this future is only available from 3.7

OR is the solution `pipx`? I only used the official installer in the past.

**Comment by radoering** [comment]
I use pipx with suffixes as described in https://python-poetry.org/docs/#installing-with-pipx

**Comment by Suprithvarma1** [comment]
by using pipx 
$ pipx install --suffix "@1.4" poetry==1.4.2
$ pipx install --suffix "@1.3" poetry==1.3.5

**Comment by MaKaNu** [comment]
Okay this explains my issues I had. The main reason I and I think many others have is the installation of a lower version if the latest is no longer supported by an older python version.

Is it somehow reasonable to control the suffix with pyenv? So as example, if I set the version of python to 3.6 in a repo dir poetry is binded to poetry version suffix I define?

**Comment by MaKaNu** [comment]
Additionally I checked that auto-completion in this case only work for the none suffix version

**Comment by dimbleby** [comment]
do whatever you like, if something around pyenv works for you then go for it.  The poetry docs provide a solution already.

this really isn't a poetry question at all, you would have the same question for any python tool where you wanted to have more than one version available: poetry is only an example of that.

**Comment by MaKaNu** [comment]
I agree. I would describe my issue from what I've learned in this discussion as following:

without pipx experience and usage of official installer (which I used and I think many first time users also) it is not very clear in the documentation that pipx is the solution to go for multiple versions of poetry.

Maybe a small hint at the beginning of the pipx installation process helps to explain when and why somebody who is not using pipx daily should tend to use this method.
